python-package/lightgbm/basic.py

```diff
@@ -1115,13 +1188,144 @@ def __init__(self, data, label=None, reference=None,
         self.feature_penalty = None
         self.monotone_constraints = None
         self.version = 0
+        self._start_row = 0  # Used when pushing rows one by one.
 
     def __del__(self):
         try:
             self._free_handle()
         except AttributeError:
             pass
 
+    def create_sample_indices(self, total_nrow):
+        """Create sample indices for the given parameter of the Dataset.
+
+        Parameters
+        ----------
+        total_nrow: int
+            Total number of rows to sample from.
+            If Dataset has multiple input data, this should be the sum of rows of every file.
+
+        Returns
+        -------
+        indices: numpy array
+            Indices for sampled data.
+        """
+        param_str = param_dict_to_str(self.params)
+        sample_cnt = self.params.get("bin_construct_sample_cnt") or DEFAULT_BIN_CONSTRUCT_SAMPLE_CNT
+        sample_cnt = min(sample_cnt, total_nrow)
+        indices = np.zeros(sample_cnt, dtype=np.int32)
+        ptr_data, _, _ = c_int_array(indices)
+
+        _safe_call(_LIB.LGBM_SampleIndices(
+            ctypes.c_int(total_nrow),
+            c_str(param_str),
+            ptr_data,
+        ))
+        return indices
+
+    def init_from_ref_dataset(self, total_nrow: int, ref_dataset: 'Dataset'):
+        """Create dataset from a reference dataset.
+
+        Args
+        ----------
```

Comment:
```suggestion
        Parameters
        ---------------
```

This project's documentation style uses `Parameters`, not `Args` for the list of parameters. You can look at the other docstrings in this file for reference.